lib/api/route/messages.dart

```diff
@@ -90,6 +92,7 @@ Future<GetMessagesResult> getMessages(ApiConnection connection, {
   bool? applyMarkdown,
   // bool? useFirstUnreadAnchor // omitted because deprecated
 }) {
+  final supportsEmptyTopics = connection.zulipFeatureLevel! >= 334; // TODO(server-10)
```

Comment:
`clientGravatar` looks like another client-capability param, but that one is exposed to the caller of this binding. I think we should maybe also expose `supportsEmptyTopics`, to make the binding as thin as possible? (Here and in other bindings)

Reply:
Ended up rewriting most of the last commit, and added a bunch of new tests.